File: src/RentCars.Domain/RentalRequests/RentalRequest.cs
using RentCars.Domain.RentalRequests.Enums;

namespace RentCars.Domain.RentalRequests;

public class RentalRequest
{
    public Guid Id { get; }
    public Guid UserId { get; }
    public Guid VehicleId { get; }
    public DateTime RentalStartDateTimeUtc { get; }
    public DateTime RentalEndDateTimeUtc { get; }
    public RentalStatus Status { get; }

    public RentalRequest(
        Guid id, Guid userId, Guid vehicleId,
        DateTime rentalStartDateTimeUtc, DateTime rentalEndDateTimeUtc,
        RentalStatus status
    )
    {
        Id = id;
        UserId = userId;
        VehicleId = vehicleId;
        RentalStartDateTimeUtc = rentalStartDateTimeUtc;
        RentalEndDateTimeUtc = rentalEndDateTimeUtc;
        Status = status;
    }
}

File: src/RentCars.WebAPI/ClientApp/src/domain/rentalRequests/rentalRequest.ts
import { RentalStatus } from "./enums/rentalStatus";

export class RentalRequest {
    constructor(
        public id: string,
        public userId: string,
        public vehicleId: string,
        //public rentalStartDateTimeUtc: null,
        //public rentalEndDateTimeUtc: null,
        public status: RentalStatus
    ) { }
}